source/clear-linux/tutorials/redis.rst

```diff
@@ -0,0 +1,254 @@
+
+      systemctl start redis
+
+   .. note::
+
+      To stop redis run :command:`systemctl stop redis`.
+
+#. Run the status command to assure the service is running:
+
+   .. code-block:: bash
+
+      redis-cli ping
+
+   .. note::
+
+      Expected output: `PONG`.
+
+#. Verify the redis-server is running.
+
+   .. code-block:: bash
+
+      systemctl status redis
+
+#. Optional: If you wish to apply advanced configuration, copy the
+   `redis.conf` into /etc/ directory.
+
+   .. code-block:: bash
+
+      sudo cp /usr/share/defaults/etc/redis.conf /etc/
+
+Example 1: Run redis-cli and try commands
+*****************************************
+
+One advantage of redis over other NoSQL databases is that developers can
+easily access data structures like lists, sets, sorted sets, strings, and
+hashes using collection operations commands similar to those found in many
+programming languages. These exercises are inspired by `try redis io`_.
+
+After your `redis-server` is running, try some basic commands.
+
+#. Enter the `redis-cli`. It provides syntax suggestions as you type.
+
+   .. code-block:: bash
+
+      redis-cli
+
+#. SET key to hold string value. In the set create connections and increment.
+
+   .. code-block:: bash
+
+      SET server:name "clearlinux"
+
+   .. code-block:: bash
+
+      MGET server:name
+
+      .. note::
+         If the key does not exist or hold a key value, `nil` is returned.
+
+   .. code-block:: bash
+
+      SET connections 100
+
+   .. code-block:: bash
+
+      INCR connections
+
+   .. code-block:: bash
+
+      INCR connections
+
+   .. code-block:: bash
+
+      DEL connections
+
+#. Create a `friends` list and insert new values at the end of the list.
+
+   .. code-block:: bash
+
+      RPUSH friends "Deb"
+
+   .. code-block:: bash
+
+      RPUSH friends "David"
+
+   .. code-block:: bash
+
+      RPUSH friends "Mary"
+
+#. Modify `friends` list, using a common slice method with a 0-index.
+
+   .. code-block:: bash
+
+      LRANGE friends 0 1
+
+   .. code-block:: bash
+
+      LLEN friends
+
+   .. code-block:: bash
+
+      LPOP friends
+
+   .. code-block:: bash
+
+      RPOP friends
+
+   .. code-block:: bash
+
+      LLEN friends
+
+#. A hash is a very useful data type that maps string fields and string
+   values, offering multiple lookup methods.
+
+   Enter many user key:values with `HMSET`. Then try `HGET` and `HGETALL`.
+
+   .. code-block:: bash
+
+      HMSET user:1000 name "Robert Noyce" password "SuperEngi9eer" email "robert.noyce@intel.com"
+
+   .. code-block:: bash
+
+      HGET user:1000 name
+
+   .. code-block:: bash
+
+      HGET user:1000 email
+
+   .. code-block:: bash
+
+      HGETALL user:1000
+
+
+Example 2: Run the |CL| redis docker image
+******************************************
+
+We also provide a `Clear Linux Redis instance`_, which is
+updated continuously and maintained by |CL| development.
+
+.. code-block:: bash
+
+   sudo swupd bundle-add cloud-control
+
+
```

Comment:
There is also an implied step here that Docker is running (i.e. `systemctl start docker`). It would b ea good idea to include that here and link back to the docker.rst document.